Refactor process_include_ids to a common base class

taldcroft opened the pull request:
This removes the class=0 restriction on included guide stars.

It also refactors acq/guide into a common base method.

Fixes #204.

jeanconn commented:
This will probably also break my guide star reporting that somewhat assumed guide stars would be fully contained within the dark current map.

jeanconn commented:
And per my comment in the issue, I think thumbs_up would need to be adjusted to at least only count guide stars that are fully in the guide-allowed FOV.

There's the other issue that I don't know if thumbs_up should just be 0  to help identify when the user does something super-weird or if we need another warning interface for the matlab tools (I think right now we're only passing exceptions back).

taldcroft commented:
Curses.  Your points are well taken, thinking.

jeanconn commented:
We've got time on this.  I don't think this needs to go out in a bugfix patch now which puts the issue at the Starcheck 13.1 level.

taldcroft commented:
I think the 90c7b61 addresses your concerns and that no other code changes would be required.  The upshot is a refactor and taking out the `CLASS` filtering.  And better test coverage.

jeanconn commented:
I haven't gone back to see which edges those are.

taldcroft commented:
The edges had matched your original test, but were not the ones you probably wanted.  I think 7c486c3 should be fine now, as it replicates the spatial test in getting initial candidates.
